infrastructure/client/api_client.go

```diff
@@ -0,0 +1,52 @@
+package client
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"taskmanager/domain/model"
+)
+
+type apiClient struct {
+	client    *http.Client
+	Token     string
+	ProjectId string
+}
+
+type ApiClient interface {
+	RequestAllTasks(t []*model.Task) ([]*model.Task, error)
+}
+
+func NewApiClient(token string, projectId string) ApiClient {
+	httpClient := &http.Client{}
+	return &apiClient{
+		client:    httpClient,
+		Token:     token,
+		ProjectId: projectId,
+	}
+}
+
+func (api *apiClient) RequestAllTasks(t []*model.Task) ([]*model.Task, error) {
+	url := "https://api.todoist.com/rest/v1/tasks"
+
+	req, _ := http.NewRequest("GET", url, nil)
```

Comment:
Remember to handle all the errors. 😄

Reply:
You are right, I followed some documentation which was mainly demonstrative, I will add the err value and its if